erigon-lib/state/domain.go

```diff
@@ -1310,9 +1310,20 @@ func buildAccessor(ctx context.Context, d *seg.Decompressor, compressed seg.File
 
 	defer d.EnableMadvNormal().DisableReadAhead()
 
+	defer func() {
+		if rec := recover(); rec != nil {
+			err = fmt.Errorf("%s %s: %s", fileName, rec, dbg.Stack())
+		}
+	}()
+
+	//allow disable compression in future:
+	detectedCompression := seg.DetectCompressType(d.MakeGetter())
```

Comment:
not sure if I understood, isn't that logic equivalent to always overriding the value with the detected one? in which case is the one passed as a parameter not overriden?